### PR TITLE
fix(schema): Fix directives cannot be extends

### PR DESCRIPTION
**Pull request by lijianzhang**
For Example:

```typescript
@ObjectType({ isAbstract: true })
class INode {
    @Field(() => ID)
    id!: string

    @Directive('@formatDate')
    @Field(() => Timestamp)
    createdAt: Date;

    @Directive('@formatDate')
    @Field(() => Timestamp)
    updatedAt: Date;
}

//  People class cannot extends INode Filed directives  
@ObjectType()
class People extends INode {

}
```

### CODE REVIEW

**Comment by lijianzhang** [comment]
@MichalLytek Thanks for your reminder. I have added test cases 😅

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
 - fix calling field resolver without providing resolver class to `buildSchema`
 - fix generated TS union type for union type of object type classes extending themselves (#587)
 - fix using shared union and interface types in multiple schemas when `resolveType` is used
+- fix `@Directive()` cannot be extends by subclasses
```
Comment (MichalLytek) [inline CHANGELOG.md:34]: This entry it's for the "unreleased" section as merging this PR won't overwrite `1.0.0-rc.1` 😉

File: tests/functional/directives.ts [hunk 2]
```diff
@@ -378,6 +397,18 @@ describe("Directives", () => {
       });
     });
 
+    describe("SubInputType", () => {
```
Comment (MichalLytek) [inline tests/functional/directives.ts:400]: it shouldn't be own describe block with type name, but rather just a `it` test case like "adds inherited field directives to input type fields while extending input type class"

File: tests/functional/directives.ts [hunk 3]
```diff
@@ -416,6 +447,45 @@ describe("Directives", () => {
         });
       });
     });
+
+    describe("SubObjectType", () => {
```
Comment (MichalLytek) [inline tests/functional/directives.ts:451]: same title issue

**Comment by lijianzhang** [comment]
@MichalLytek  I have dealt with it. thank you for your patience to point out these problems

and `type-graphql` is awesome.

**Pull request merged**